robocop/checkers/spacing.py

```diff
@@ -123,3 +124,28 @@ def visit_File(self, node):  # noqa
                 self.report("empty-lines-between-sections", empty_lines, self.empty_lines_between_sections,
                             lineno=section.end_lineno, col=0)
         super().visit_File(node)
+
+
+class InconsistentUseOfTabsAndSpacesChecker(VisitorChecker):
+    """ Checker for inconsistent use of tabs and spaces. """
+
+    rules = {
+        "1006": (
+            "inconsistent-use-of-tabs-and-spaces",
+            "Inconsistent use of tabs and spaces in file",
+            RuleSeverity.WARNING
+        )
+    }
+
+    def visit_File(self, node):  # noqa
+        tabs = False
+        spaces = False
+
+        for token in get_tokens(node.source):
+            if token.type != 'SEPARATOR':
+                continue
+            tabs = True if '\t' in token.value else tabs
+            spaces = True if ' ' in token.value else spaces
+
+        if tabs and spaces:
```

Comment:
Shouldn't this be inside of `for` loop to decrease the number of rotations? We can break the `for` loop if this is true and no need to run it through all tokens.

Reply:
Totally agree!